<launch>
    <arg name="rand" default="10.0" />
    <arg name="num" default="5" />
    <arg name="detection" default="true" />
    <arg name="display" default="false" />
    
    <node name="swarm_loop" pkg="swarm_loop" type="swarm_loop_node" output="log">
        <rosparam>
            lcm_uri: udpm://224.0.0.251:7667?ttl=1
            self_id: 0
        </rosparam>
       <param name="debug_image" value="$(arg display)" type="bool" />
       <param name="camera_config_path" value="$(env HOME)/swarm_ws/src/VINS-Fusion-gpu/config/vi_car/cam0_mei.yaml" type="string" />

        <remap from="~left_camera" to="/cam0/image_raw"/>
    </node>
    <node name="vins_estimator" pkg="vins" type="vins_node" output="log" args="$(env HOME)/swarm_ws/src/VINS-Fusion-gpu/config/vi_car/vi_car.yaml" />
    
    <node pkg="localization_proxy" name="localization_proxy" type="localization_proxy_node" output="log">
       <param name="vins_topic" value="/vins_estimator/imu_propagate" type="string" />
       <rosparam>
            send_fused_freq: 0.0
            send_fused_basecoor_freq: 30.0
            self_id: 0
            enable_uwb: false
       </rosparam>
    </node>
    <node pkg="swarm_localization" name="swarm_localization" type="swarm_localization_node" output="log">
        <rosparam>
            force_freq: 1
            max_accept_cost: 10000
            max_keyframe_num: 1000
            min_keyframe_num: 3
            thread_num: 2
            min_kf_movement : 0.5
            init_xy_movement : 1.0
            init_z_movement : 0
            pub_swarm_odom: true
        </rosparam>
        <param name="swarm_nodes_config" value="$(find swarm_localization)/config/swarm_nodes5.yaml" type="string" />
        <param name="initial_random_noise" value="$(arg rand)" type="double" />
    </node>
    <node name="my_desc_server" pkg="tx2_whole_image_desc_server" type="py2_whole_image_desc_server_ts.py" output="screen">
        <param name="frozen_protobuf_file" type="string" value="$(find tx2_whole_image_desc_server)/tx2_py3_tfserver/models.keras/June2019/centeredinput-m1to1-240x320x1__mobilenetv2-block_9_add__K16__allpairloss/modelarch_and_weights.2000.480x640x1.pb" />
        <param name="nrows" type="string" value="480" />
        <param name="ncols" type="string" value="640" />
        <param name="nchnls" type="string" value="1" />
        <remap from="left_camera" to="/cam0/image_raw"/>
    </node>

    <node pkg="swarm_yolo" name="swarm_yolo" type="detect.py" output="screen">
        <param name="image_topic" type="string" value="/camera/infra1/image_rect_raw" />
		<param name="model_def" type="string" value="$(find swarm_yolo)/config/yolov3-tiny-1class.cfg" />
		<param name="weights_path" type="string" value="$(find swarm_yolo)/weights/yolov3-tiny_drone.pth" />
		<param name="weights_trt_path" type="string" value="$(find swarm_yolo)/weights/yolov3-tiny_drone_trt.pth" />
		<param name="class_path" type="string" value="$(find swarm_yolo)/config/drone.names" />
		<param name="debug_show" type="string" value="true" />
		<rosparam>
            conf_thres: 0.6
        </rosparam>
	</node>
</launch>
